src/lib/routes/client-api/feature.ts

```diff
@@ -130,39 +119,30 @@ export default class FeatureController extends Controller {
             ],
         });
 
-        if (clientFeatureCaching?.enabled) {
-            this.cache = true;
-            this.cachedFeatures = memoizee(
-                (query) => this.resolveFeaturesAndSegments(query),
+        if (clientFeatureCaching.enabled) {
+            this.featuresAndSegments = memoizee(
```

Comment:
This one uses the code of `cachedFeatures2` from the old code, but instead of defining a field holding the cache, we're dynamically defining the behavior of the method at construction time, to either use the cache or call directly to the method